checkpoint/util.go

```diff
@@ -0,0 +1,53 @@
+package checkpoint
+
+import (
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+
+	"github.com/santhosh-tekuri/jsonschema/v5"
+	"github.com/spf13/afero"
+
+	"github.com/spacemeshos/go-spacemesh/log"
+)
+
+func ValidateSchema(data []byte) error {
+	sch, err := jsonschema.CompileString(schemaFile, Schema)
+	if err != nil {
+		return fmt.Errorf("compile checkpoint json schema: %w", err)
+	}
+	var v any
+	if err = json.Unmarshal(data, &v); err != nil {
+		return fmt.Errorf("unmarshal checkpoint data: %w", err)
+	}
+	if err = sch.Validate(v); err != nil {
+		return fmt.Errorf("validate checkpoint data: %w", err)
+	}
+	return nil
+}
+
+func savefile(logger log.Log, fs afero.Fs, dst string, data []byte) error {
```

Comment:
do you have a ballpark estimate how large snaphot can be? it would probably make more sense to stream data into a file, instead of holding full blob in memory

Reply:
assuming 10K atx and 20K accounts (each holding 5KB state) in the network, we are looking at 
400B/atx and 10KB/account. that will be
8MB (2 atx per miner) +200MB = 208MB

yes. i will change it to streaming.

Reply:
account definitely can't be 5kb even in json. maybe around 1kb is max

Reply:
changed to streaming directly write to a file.
tho not so much streaming. i wasn't able to figure out how to write json's structured data in chunks.
ended up creating a struct that holds all the data, and encode it directly to a file.

please let me know if there is a better approach.

Reply:
yes it is not exactly what i meant, you still hold full checkpoint in memory. json probably will not allocate memory for whole checkpoint. but otherwise it is still not optimal

in order to stream checkpoint you essentially need to write incrementally, not sure if standard codec supports that, so the program should do smth like

```
fmt.Println("{")
fmt.Println('"layer": 10')
fmt.Printn('atxs: [')
for item in atxs {
     fmt.Println(item.JSONString())
}
```

there should be lib that can do that, but i don't know if needed

Reply:
filed https://github.com/spacemeshos/go-spacemesh/issues/4379 to follow up